SCClassLibrary/Common/Core/CondVar.sc

```diff
@@ -0,0 +1,105 @@
+CondVar {
+	var waitingThreads, timeoutThreads;
+
+	*new { ^super.newCopyArgs(Array.new, Array.new) }
+
+	wait { |predicate|
```

Comment:
This will loop "full speed" until the predicate returns true. Is that absolutely required? It seems risky and one needs to know what one is doing.

Isn't `signal` supposed to make such a while loop unnecessary`

Reply:
I think you might be misunderstanding -- this isn't a busy loop; each `wait` call inside the loop blocks until woken via signalling, and only when the thread is woken from a signal does it check the predicate again. If it's true, we exit the loop and return from the wait; otherwise, we go back to waiting.

Reply:
ah yes, now I can spot the trick, the inner wait doesn't pass the predicate so that it'll sleep. Intricate :)

Reply:
both you and James had trouble understanding this, so i'm planning to add a second private function which both branches will call down to, and do the same for waitFor. I think that will make it easier to understand